cli/catalog-api-v1/build.rs

```diff
@@ -0,0 +1,39 @@
+use std::fs;
+use std::path::PathBuf;
+
+use openapiv3::OpenAPI;
+
+fn main() {
+    let manifest_dir = PathBuf::from(std::env::var("CARGO_MANIFEST_DIR").unwrap());
+    let out_dir = PathBuf::from(std::env::var("OUT_DIR").unwrap());
+
+    let spec_src = manifest_dir.join("openapi.json");
+
+    println!("cargo:rerun-if-changed={}", spec_src.display());
+
+    let file = std::fs::File::open(spec_src).unwrap();
+    let spec = serde_json::from_reader(file).expect("Failed to parse openapi spec");
+
+    let client = generate_client(&spec);
+    let client_dst = out_dir.join("client.rs");
+    fs::write(client_dst, client).unwrap();
+
+    let mock = gnereate_mock(&spec);
+    let mock_dst = out_dir.join("mock.rs");
+    fs::write(mock_dst, mock).unwrap();
+}
+
+fn generate_client(spec: &OpenAPI) -> String {
+    let mut generator = progenitor::Generator::default();
+    let tokens = generator.generate_tokens(spec).unwrap();
+    let ast = syn::parse2(tokens).unwrap();
+    prettyplease::unparse(&ast)
+}
+
+fn gnereate_mock(spec: &OpenAPI) -> String {
```

Comment:
```suggestion
fn generate_mock(spec: &OpenAPI) -> String {
```

Reply:
Well it'd always go throught he build.rs, i thought w decided on just generating it for to keep updating simple, but generating the files into the source dir also works 🤷🏼‍♂️

Reply:
I remember a different conversation. I remember us deciding to generate the code and check it in so that we can always view the code. Although I can't remember if the `httpmock` can be generated that way.

Reply:
I don't have strong opinions, but that is what I remember and put in the ticket